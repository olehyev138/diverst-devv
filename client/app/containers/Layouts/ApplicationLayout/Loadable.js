/**
 *
 * Asynchronously loads the component for ApplicationLayout
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
