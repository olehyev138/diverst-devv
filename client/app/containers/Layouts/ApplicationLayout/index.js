import React, { memo, useEffect } from 'react';
import { Route } from 'react-router';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import CssBaseline from '@material-ui/core/CssBaseline';

const styles = theme => ({});

export const RouteContext = React.createContext(null);

const ApplicationLayout = ({ component: Component, ...rest }) => {
  const { classes, ...other } = rest;

  return (
    <Route
      {...other}
      render={routeProps => (
        <RouteContext.Provider
          value={{
            computedMatch: routeProps.match,
            location: routeProps.location
          }}
        >
          <CssBaseline />
          <Component {...rest} />
        </RouteContext.Provider>
      )}
    />
  );
};

ApplicationLayout.propTypes = {
  classes: PropTypes.object,
  component: PropTypes.elementType,
};

export default withStyles(styles)(ApplicationLayout);
