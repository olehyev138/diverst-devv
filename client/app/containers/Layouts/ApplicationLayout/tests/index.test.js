import React from 'react';
import PropTypes from 'prop-types';
import { shallow } from 'enzyme';
import { unwrap } from '@material-ui/core/test-utils';

import ApplicationLayout from '../index';

/**
 * TODO:
 *  - test it accepts & renders a child component
 */


const props = {
  classes: {},
  component: jest.fn()
};

describe('<ApplicationLayout />', () => {
  ApplicationLayout.contextTypes = {
    router: PropTypes.object
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<ApplicationLayout {...props} />);

    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<ApplicationLayout {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
