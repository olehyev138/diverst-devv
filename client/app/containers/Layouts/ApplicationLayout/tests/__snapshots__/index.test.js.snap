// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationLayout /> Should render and match the snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
